### PR TITLE
Bonzo finance

### PR DESCRIPTION
**Pull request by bugbytesinc**
Thankyou for accepting our general adapter PR, here is a proposed dimension adapter PR

---
##### Name (to be shown on DefiLlama):
Bonzo Finance

##### Twitter Link:
https://x.com/bonzo_finance

##### List of audit links if any:
https://docs.bonzo.finance/hub/resources/audits

##### Website Link:
https://bonzo.finance/

##### Current TVL:
0

##### Treasury Addresses (if the protocol has treasury)
0x00000000000000000000000000000000005dc4d4

##### Chain:
295

##### Short Description (to be shown on DefiLlama):
The Liquidity Layer of Hedera: An open source, non-custodial lending protocol based on Aave.

##### Category (full list at https://defillama.com/categories) *Please choose only one:
Lending

##### Oracle Provider(s): Specify the oracle(s) used (e.g., Chainlink, Band, API3, TWAP, etc.):
Supra Oracle

##### Implementation Details: Briefly describe how the oracle is integrated into your project:
Bonzo Finance uses Supra’s push oracles; these oracles feed asset price information at regular intervals to the Hedera network and enable the Bonzo Finance protocol to query the most recent or historical data.

##### Documentation/Proof: Provide links to documentation or any other resources that verify the oracle's usage:
https://docs.bonzo.finance/hub/developer/oracles/supra 

##### forkedFrom (Does your project originate from another project):
Aave

##### Github org/user (Optional, if your code is open source, we can track activity):
https://github.com/Bonzo-Labs

### CODE REVIEW

**Comment by llamatester** [comment]
The bonzo adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts fees bonzo
    
    🦙 Running BONZO adapter 🦙
    ---------------------------------------------------
    Start Date:	Sun, 27 Oct 2024 00:00:00 GMT
    End Date:	Mon, 28 Oct 2024 00:00:00 GMT
    ---------------------------------------------------
    
    HEDERA 👇
    Backfill start time: 28/10/2024
    End timestamp: 1730119513.15844 (2024-10-28T12:45:13.158Z)
    Daily fees: 0
    └─ Methodology: Intrest and Flash Loan fees plus liquidation bonuses in USD
    Daily user fees: 0
    └─ Methodology: Intrest and Flash Loan fees plus liquidation bonuses in USD
    Daily protocol revenue: 0
    └─ Methodology: Portion of intrest rate fees and flash loan fees to Protocol Treasury in USD
    Daily supply side revenue: 0
    └─ Methodology: Portion of intrest rate fees to liquidity providers in USD

File: fees/bonzo/index.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import BigNumber from "bignumber.js";
+import { httpGet } from "../../utils/fetchURL";
+
+const fetch = async () => {
+    const res = await httpGet("https://bonzo-data-api-eceac9d8a2aa.herokuapp.com/stats");
+    const timestamp = parseFloat(res.timestamp_end);
+    // Portion of intrest payments sent to the protocol over the 24hr period.
+    const total_protocol_fees = BigNumber(res.total_protocol_fees.usd_wad);
```
Comment (dtmkeng) [inline fees/bonzo/index.ts:8]: value is it daily value ?

**Review by dtmkeng — CHANGES_REQUESTED** [review]

File: fees/bonzo/index.ts (hunk 1 above)
Comment (bugbytesinc) [reply fees/bonzo/index.ts:8]: Yes, if I understand your definition correctly.

It is the USD table value of the sum of the protocol fees sent to the protocol's treasury wallet for the past 24 hours from the time of call.

Comment (dtmkeng) [reply fees/bonzo/index.ts:8]: why it still 0 ?

Comment (bugbytesinc) [reply fees/bonzo/index.ts:8]: The protocol has not yet turned on borrowing, so there is no income generated yet.

**Comment by bugbytesinc** [comment]
Hello, thankyou for merging our PR back in October.  We have enabled borrowing on our site, and when I test locally I receive the following output:

```
$ npm test fees bonzo

> adapters@1.0.0 test
> ts-node --transpile-only cli/testAdapter.ts fees bonzo

🦙 Running BONZO adapter 🦙
---------------------------------------------------
Start Date:     Wed, 01 Jan 2025 00:00:00 GMT
End Date:       Thu, 02 Jan 2025 00:00:00 GMT
---------------------------------------------------

HEDERA 👇
Backfill start time: 2/1/2025
End timestamp: 1735849266.8877 (2025-01-02T20:21:06.887Z)
Daily fees: 561
└─ Methodology: Intrest and Flash Loan fees plus liquidation bonuses in USD
Daily user fees: 561
└─ Methodology: Intrest and Flash Loan fees plus liquidation bonuses in USD
Daily protocol revenue: 66
└─ Methodology: Portion of intrest rate fees and flash loan fees to Protocol Treasury in USD
Daily supply side revenue: 495
└─ Methodology: Portion of intrest rate fees to liquidity providers in USD
```

However, when I go to information page for bonzo finance:

https://defillama.com/protocol/bonzo-finance

In the methodology section I do not see a link to "Fees and Revenue code", which I would have expected to take me to our adapter code. 

Is there something we might have missed or need to fix to ensure it displays correctly?

Would it be beneficial to start a new PR for this discussion (there are some typos to fix on our part as well)

Thank you for your time and guidance.

Bonzo Team

**Pull request merged**